modules/subchannel/src/problems/TriInterWrapper1PhaseProblem.C

```diff
@@ -0,0 +1,605 @@
+  }
+
+  // Initialize an average pressure drop for uniform pressure inlet condition
+  auto dPdZ = dPdZ_i.mean();
+  auto error = 1.0;
+  auto tol = 1e-6;
+  auto iter = 0;
+  auto max_iter = 10;
+  while (error > tol)
+  {
+    iter += 1;
+    if (iter > max_iter)
+      mooseError(name(), ": exceeded maximum number of iterations");
+    auto dPdZ_old = dPdZ;
+    for (unsigned int i_ch = 0; i_ch < _subchannel_mesh.getNumOfChannels(); i_ch++)
+    {
+      // update inlet mass flow to achieve DPDZ
+      mass_flow_i(i_ch) = computeMassFlowForDPDZ(dPdZ, i_ch);
+    }
+    // Calculate total massflow at the inlet
+    auto mass_flow_sum = mass_flow_i.sum();
+    // Update the DP/DZ to correct the mass flow rate.
+    dPdZ *= std::pow(total_mass_flow / mass_flow_sum, 2.0);
+    error = std::abs((dPdZ - dPdZ_old) / dPdZ_old);
+  }
+
+  // Populate solution vector with corrected boundary conditions
+  for (unsigned int i_ch = 0; i_ch < _subchannel_mesh.getNumOfChannels(); i_ch++)
+  {
+    auto * node = _subchannel_mesh.getChannelNode(i_ch, 0);
+    _mdot_soln->set(node, mass_flow_i(i_ch));
+  }
+  _console << "Done applying mass flow boundary condition\n";
+}
+
+void
+TriInterWrapper1PhaseProblem::computeInletMassFlowDist()
+{
+  auto mass_flow = 0.0;
+  // inlet pressure
+  Eigen::VectorXd p_in(_subchannel_mesh.getNumOfChannels());
+  // outlet pressure
+  Eigen::VectorXd p_out(_subchannel_mesh.getNumOfChannels());
+  // pressure drop
+  Eigen::VectorXd dpz(_subchannel_mesh.getNumOfChannels());
+  // inlet mass flow rate
+  Eigen::VectorXd mass_flow_i(_subchannel_mesh.getNumOfChannels());
+  // inlet mass flow flux
+  Eigen::VectorXd mass_flux_i(_subchannel_mesh.getNumOfChannels());
+  // new/corrected mass flux
+  Eigen::VectorXd g_new(_subchannel_mesh.getNumOfChannels());
+  // flow area
+  Eigen::VectorXd Si(_subchannel_mesh.getNumOfChannels());
+  // total number of subchannels
+  auto tot_chan = _subchannel_mesh.getNumOfChannels();
+  // average pressure
+  auto dpz_ave = 0.0;
+  // summmation for the pressure drop over all subchannels
+  auto dpzsum = 0.0;
+  // new total mass flow rate - used for correction
+  auto mass_flow_new_tot = 0.0;
+  // number of axial nodes
+  unsigned int nz = _subchannel_mesh.getNumOfAxialCells();
+
+  for (unsigned int i_ch = 0; i_ch < _subchannel_mesh.getNumOfChannels(); i_ch++)
+  {
+    auto * node = _subchannel_mesh.getChannelNode(i_ch, 0);
+    auto * node_out = _subchannel_mesh.getChannelNode(i_ch, nz);
+    Si(i_ch) = (*_S_flow_soln)(node);
+    mass_flow_i(i_ch) = (*_mdot_soln)(node);
+    mass_flux_i(i_ch) = mass_flow_i(i_ch) / Si(i_ch);
+    p_in(i_ch) = (*_P_soln)(node);
+    p_out(i_ch) = (*_P_soln)(node_out);
+    dpz(i_ch) = p_in(i_ch) - p_out(i_ch);
+    if (dpz(i_ch) <= 0.0)
+    {
+      mooseError(
+          name(), " Computed presurre drop at the following subchannel is less than zero. ", i_ch);
```

Comment:
```suggestion
          name(), " Computed pressure drop at the following subchannel is less than zero. ", i_ch);
```
so this problem cannot handle reversed flow?